sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs

```diff
@@ -1110,8 +1112,9 @@ public partial class PostgreSqlDataFeedSource : Azure.AI.MetricsAdvisor.Models.D
     public partial class ServicePrincipalDatasourceCredential : Azure.AI.MetricsAdvisor.Models.DatasourceCredential
     {
         public ServicePrincipalDatasourceCredential(string name, string clientId, string clientSecret, string tenantId) { }
-        public string ClientId { get { throw null; } }
-        public string TenantId { get { throw null; } }
+        public string ClientId { get { throw null; } set { } }
+        public string TenantId { get { throw null; } set { } }
+        public void UpdateClientSecret(string clientSecret) { }
```

Comment:
Adding setters and an "update secret" method so users can easily update this type using our Update APIs.

Reply:
The `UpdateClientSecret` method follows the same pattern we use in our key credentials.